### PR TITLE
Test/scanner location

### PR DESCRIPTION
**Pull request by goofiva**
Profile: forseti_integration_testing
Version: (not specified)
Target:  ssh://ubuntu@10.128.15.203:22

  ✔  scanner-location: Command: `sudo gsutil cp -r gs://forseti-server-cf8f24ea/rules /home/ubuntu/forseti-security`
     ✔  Command: `sudo gsutil cp -r gs://forseti-server-cf8f24ea/rules /home/ubuntu/forseti-security` exit_status should eq 0
     ✔  Command: `sudo systemctl restart config-validator` exit_status should eq 0
     ✔  Command: `forseti inventory create --import_as 3604a1a551a` exit_status should eq 0
     ✔  Command: `forseti inventory create --import_as 3604a1a551a` stdout should match /"id": "([0-9]*)"/
     ✔  Command: `forseti model use 3604a1a551a` exit_status should eq 0
     ✔  Command: `forseti scanner run` exit_status should eq 0
     ✔  Command: `forseti scanner run` stdout should match /Scanner Index ID: (.*[0-9].*) is created/
     ✔  Command: `mysql -u forseti_security_user -p{PASSWORD} --host 127.0.0.1 --execute "SELECT COUNT(*) FROM forseti_security.violations V JOIN forseti_security.scanner_index SI ON SI.id = V.scanner_index_id WHERE SI.inventory_index_id = 1579134138034067 AND V.violation_type = 'LOCATION_VIOLATION' AND V.rule_name = 'All buckets in project must not be in EU';"` exit_status should eq 0
     ✔  Command: `mysql -u forseti_security_user -p{PASSWORD} --host 127.0.0.1 --execute "SELECT COUNT(*) FROM forseti_security.violations V JOIN forseti_security.scanner_index SI ON SI.id = V.scanner_index_id WHERE SI.inventory_index_id = 1579134138034067 AND V.violation_type = 'LOCATION_VIOLATION' AND V.rule_name = 'All buckets in project must not be in EU';"` stdout should match /2/

Profile Summary: 1 successful control, 0 control failures, 0 controls skipped
Test Summary: 9 successful, 0 failures, 0 skipped
       Finished verifying <forseti-local> (27m46.37s).
-----> Kitchen is finished. (27m48.50s)

### CODE REVIEW

File: integration_tests/tests/forseti/controls/scanner-location.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+control 'scanner-location' do
+  # Arrange
+  # Copy the constraints from GCS
+  describe command("sudo gsutil cp -r gs://#{forseti_server_storage_bucket}/rules /home/ubuntu/forseti-security") do
+    its('exit_status') { should eq 0 }
+  end
+
+  # Restart Config Validator
+  describe command("sudo systemctl restart config-validator") do
```
Comment (gkowalski-google) [inline integration_tests/tests/forseti/controls/scanner-location.rb]: This is not needed, this scanner is not related to CV.

File: integration_tests/tests/forseti/controls/scanner-location.rb [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    its('stdout') { should match /"id": "([0-9]*)"/}
+  end
+  @inventory_id = /"id": "([0-9]*)"/.match(inventory_create.stdout)[1]
+
+  describe command("forseti model use #{model_name}") do
+    its('exit_status') { should eq 0 }
+  end
+
+  # Act
+  describe command("forseti scanner run") do
```
Comment (gkowalski-google) [inline integration_tests/tests/forseti/controls/scanner-location.rb]: It's safer to run forseti model use and forseti scanner run in the same command, otherwise sometimes the scanner command gets run first `forseti model use #{model_name} && forseti scanner run`.

File: integration_tests/tests/forseti/controls/scanner-location.rb [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    its('exit_status') { should eq 0 }
+    its('stdout') { should match(/Scanner Index ID: (.*[0-9].*) is created/) }
+  end
+
+  # Assert Firewall violation found
+  describe command("mysql -u #{db_user_name} -p#{db_password} --host 127.0.0.1 --execute \"SELECT COUNT(*) FROM forseti_security.violations V JOIN forseti_security.scanner_index SI ON SI.id = V.scanner_index_id WHERE SI.inventory_index_id = #{@inventory_id} AND V.violation_type = 'LOCATION_VIOLATION' AND V.rule_name = 'All buckets in project must not be in EU';\"") do
+    its('exit_status') { should eq 0 }
+    its('stdout') { should match(/2/) }
+  end
```
Comment (gkowalski-google) [inline integration_tests/tests/forseti/controls/scanner-location.rb]: Delete the inventory/model at the end.

File: integration_tests/tests/forseti/controls/scanner-location.rb [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+require 'json'
+require 'securerandom'
+
+db_password = attribute('forseti-cloudsql-password')
+db_user_name = attribute('forseti-cloudsql-user')
+forseti_server_storage_bucket = attribute('forseti-server-storage-bucket')
+model_name = SecureRandom.uuid.gsub!('-', '')[0..10]
+
+control 'scanner-location' do
```
Comment (gkowalski-google) [inline integration_tests/tests/forseti/controls/scanner-location.rb]: (minor) Can you rename this file/control to be `scanner-location_scanner` to align with the naming convention for the other scanner tests?

File: integration_tests/tests/forseti/controls/scanner-location.rb [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+  # Arrange
+  # Copy the constraints from GCS
+  describe command("sudo gsutil cp -r gs://#{forseti_server_storage_bucket}/rules /home/ubuntu/forseti-security") do
```
Comment (gkowalski-google) [inline integration_tests/tests/forseti/controls/scanner-location.rb]: The hard-coded path for forseti can be replaced with the env var `$FORSETI_HOME/`

**Comment by goofiva** [comment]
@gkowalski-google addressed all comments. Please review.

Profile: forseti_integration_testing
Version: (not specified)
Target:  ssh://ubuntu@10.128.15.203:22

  ✔  scanner-location-scanner: Command: `sudo gsutil cp -r gs://forseti-server-cf8f24ea/rules $FORSETI_HOME/`
     ✔  Command: `sudo gsutil cp -r gs://forseti-server-cf8f24ea/rules $FORSETI_HOME/` exit_status should eq 0
     ✔  Command: `forseti inventory create --import_as 44eeb88d36c` exit_status should eq 0
     ✔  Command: `forseti inventory create --import_as 44eeb88d36c` stdout should match /"id": "([0-9]*)"/
     ✔  Command: `forseti model use 44eeb88d36c && forseti scanner run` exit_status should eq 0
     ✔  Command: `forseti model use 44eeb88d36c && forseti scanner run` stdout should match /Scanner Index ID: (.*[0-9].*) is created/
     ✔  Command: `mysql -u forseti_security_user -pwbu9Pk7VbLrqo9gG --host 127.0.0.1 --execute "SELECT COUNT(*) FROM forseti_security.violations V JOIN forseti_security.scanner_index SI ON SI.id = V.scanner_index_id WHERE SI.inventory_index_id = 1579205924386022 AND V.violation_type = 'LOCATION_VIOLATION' AND V.rule_name = 'All buckets in project must not be in EU';"` exit_status should eq 0
     ✔  Command: `mysql -u forseti_security_user -pwbu9Pk7VbLrqo9gG --host 127.0.0.1 --execute "SELECT COUNT(*) FROM forseti_security.violations V JOIN forseti_security.scanner_index SI ON SI.id = V.scanner_index_id WHERE SI.inventory_index_id = 1579205924386022 AND V.violation_type = 'LOCATION_VIOLATION' AND V.rule_name = 'All buckets in project must not be in EU';"` stdout should match /2/
     ✔  Command: `forseti inventory delete 1579205924386022` exit_status should eq 0
     ✔  Command: `forseti model delete 44eeb88d36c` exit_status should eq 0


Profile Summary: 1 successful control, 0 control failures, 0 controls skipped
Test Summary: 9 successful, 0 failures, 0 skipped
       Finished verifying <forseti-local> (28m4.89s).
-----> Kitchen is finished. (28m7.03s)

**Comment by gkowalski-google** [comment]
@goofiva The last build failed this test, no violations were found:

```
×  Command: `mysql -u forseti_security_user -p{PASSWORD} --host 127.0.0.1 --execute "SELECT COUNT(*) FROM forseti_security.violations V JOIN forseti_security.scanner_index SI ON SI.id = V.scanner_index_id WHERE SI.inventory_index_id = 1579213785546318 AND V.violation_type = 'LOCATION_VIOLATION' AND V.rule_name = 'All buckets in project must not be in EU';"` stdout should match /2/
     expected "COUNT(*)\n0\n" to match /2/
     Diff:
     @@ -1,2 +1,3 @@
     -/2/
     +COUNT(*)
     +0
```

**Comment by goofiva** [comment]
forseti-server: Verifying host 10.128.15.206
Skipping profile: 'inspec-gcp' on unsupported platform: 'ubuntu/18.04'.
.........

Profile: forseti_integration_testing
Version: (not specified)
Target:  ssh://ubuntu@10.128.15.206:22

  ✔  scanner-location-scanner: Command: `sudo gsutil cp -r gs://forseti-server-c2807c58/rules $FORSETI_HOME/`
     ✔  Command: `sudo gsutil cp -r gs://forseti-server-c2807c58/rules $FORSETI_HOME/` exit_status should eq 0
     ✔  Command: `forseti inventory create --import_as 72e35b0f254` exit_status should eq 0
     ✔  Command: `forseti inventory create --import_as 72e35b0f254` stdout should match /"id": "([0-9]*)"/
     ✔  Command: `forseti model use 72e35b0f254 && forseti scanner run` exit_status should eq 0
     ✔  Command: `forseti model use 72e35b0f254 && forseti scanner run` stdout should match /Scanner Index ID: (.*[0-9].*) is created/
     ✔  Command: `mysql -u forseti_security_user -peN4mDuoz7RjMdIV4 --host 127.0.0.1 --execute "SELECT COUNT(*) FROM forseti_security.violations V JOIN forseti_security.scanner_index SI ON SI.id = V.scanner_index_id WHERE SI.inventory_index_id = 1579225271889284 AND V.violation_type = 'LOCATION_VIOLATION' AND V.rule_name = 'All buckets in project must not be in EU';"` exit_status should eq 0
     ✔  Command: `mysql -u forseti_security_user -peN4mDuoz7RjMdIV4 --host 127.0.0.1 --execute "SELECT COUNT(*) FROM forseti_security.violations V JOIN forseti_security.scanner_index SI ON SI.id = V.scanner_index_id WHERE SI.inventory_index_id = 1579225271889284 AND V.violation_type = 'LOCATION_VIOLATION' AND V.rule_name = 'All buckets in project must not be in EU';"` stdout should match /2/
     ✔  Command: `forseti inventory delete 1579225271889284` exit_status should eq 0
     ✔  Command: `forseti model delete 72e35b0f254` exit_status should eq 0


Profile Summary: 1 successful control, 0 control failures, 0 controls skipped
Test Summary: 9 successful, 0 failures, 0 skipped
       Finished verifying <forseti-local> (38m57.81s).
-----> Kitchen is finished. (39m0.01s)

**Comment by goofiva** [comment]
@gkowalski-google See previous comment. I'm unable to replicate the error in a fresh environment. Any ideas?

**Comment by gkowalski-google** [comment]
Location scanner test is still failing on GCB, not reproduced locally.. wait until inventory bugs are resolved and then come back to this.
